src/lib_gui/qt/element/QtTabBar.cpp

```diff
@@ -19,3 +23,22 @@ QSize QtTabBar::minimumTabSizeHint(int index) const
 {
 	return QSize(45, QTabBar::minimumTabSizeHint(index).height());
 }
+
+void QtTabBar::contextMenuEvent(QContextMenuEvent* event)
+{
+    QtContextMenu menu(event, this);
```

Comment:
Your whole implementation uses the wrong indentation. Please change everything to `tabs` instead of `spaces`.